ehr/src/org/labkey/ehr/utils/TriggerScriptHelper.java

```diff
@@ -2256,11 +2256,20 @@ public String verifyProtocolCounts(final String id, Integer project, final List<
 
         //find the total animals previously used by this protocols/species
         TableInfo ti = QueryService.get().getUserSchema(getUser(), getContainer(), "ehr").getTable("protocolTotalAnimalsBySpecies");
-        SimpleFilter filter = new SimpleFilter(FieldKey.fromString("species"), PageFlowUtil.set(ar.getSpecies(), "All Species"), CompareType.IN);
+        String animalSpecies = ar.getSpecies();
+        SimpleFilter filter;
+        if (("Rhesus".equals(animalSpecies) || "Cynomolgus".equals(animalSpecies)))
+        {
+             filter = new SimpleFilter(FieldKey.fromString("species"), PageFlowUtil.set(animalSpecies, "All Species", "Macaque"), CompareType.IN);
+         } else
+         {
+             filter = new SimpleFilter(FieldKey.fromString("species"), PageFlowUtil.set(animalSpecies, "All Species"), CompareType.IN);
+         }
         filter.addCondition(FieldKey.fromString("protocol"), protocol);
         TableSelector ts = new TableSelector(ti, filter, null);
         final List<String> errors = new ArrayList<>();
         final String ALL_SPECIES = "All Species";
+        final boolean[] noSpeciesListedOnProtocol = {true};
```

Comment:
This is a minor comment and there's nothing wrong with using an array, but I would say AtomicBoolean is a little clearer. In my opinion is makes it more obvious what the purpose of using that object it, relative to making an array purely for the purpose of using a java quirk.